Feature/expose graphql types

paulpopus opened the pull request:
## Description

Related to https://github.com/payloadcms/payload/issues/2557 , I will update the documentation around GraphQL after this PR in case there are any further changes needed in this code.

This PR exports:
- buildPaginatedListType
- the internally used GraphQL library and type to make sure there's no conflicts (happy to revert on this one but I thought it might make the DX easier)
- the mutations and queries type into a re-usable type `GraphQLExtension` in the config

And it attaches the created `buildPaginatedType` from collections to the available types for re-use on `payload.collections.graphQL.`

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

<!-- Please delete options that are not relevant. -->

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (will come in a PR after this one)

denolfe commented on graphql.d.ts:
```diff
@@ -0,0 +1,2 @@
+export { default as buildPaginatedListType } from "./dist/graphql/schema/buildPaginatedListType";
+export {default as GraphQL} from 'graphql';
```
Let's clean up the spacing and quote type here.

denolfe commented on src/collections/graphql/init.ts:
```diff
@@ -95,10 +95,15 @@ function initCollectionsGraphQL(payload: Payload): void {
       singularName,
       whereInputFields,
-      singularName,
+      singularName
```
Add back comma

denolfe commented on graphql.js:
```diff
@@ -0,0 +1,4 @@
+exports.buildPaginatedListType =
+  require("./dist/graphql/schema/buildPaginatedListType").default;
+
+exports.GraphQL = require("graphql");
```
Discussed this with the team, we're okay with this 👍

denolfe reviewed (changes requested):
Just some formatting tweaks. Other than that looks good. 

Will re-review when docs are added 👍

paulpopus commented:
@denolfe I've fixed those code style issues. Added documentation, let me know how it looks.

Going through it again, there's probably another round of work to do to make working with GraphQL feel more first class.
